api/custom_auth/views.py

```diff
@@ -51,3 +54,22 @@ def delete_token(request):
         sender=request.user.__class__, request=request, user=request.user
     )
     return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class CustomUserDeleteView(UserViewSet):
+    def destroy(self, request, *args, **kwargs):
+        if "delete_orphan_organizations" in request.query_params:
```

Comment:
We can use the walrus operator here to simplify things. 

https://docs.python.org/3/whatsnew/3.8.html#assignment-expressions

Reply:
We can actually also offload a lot of this logic to a query serializer. There are a few examples of this throughout the code. You basically just need to define a serializer like: 

```python
class UserDeleteQuerySerializer(serializers.Serializer):
    delete_orphan_organisations = serializers.BooleanField(default=False)
```

then in the view code, you can just do: 

```python
serializer = UserDeleteQuerySerializer(data=request.query_params)
serializer.is_valid(raise_exception=True)
if serializer.data["delete_orphan_organisations"]:
     # handle the organisation deletes
super().destroy(...)
```

Reply:
Thank you @matthewelwell, I'll try using the serializer to take care of the param validation. I think it should inherit from UserDeleteSerializer since that one already inherits from CurrentPasswordSerializer.

Reply:
Custom view optimized. I made it part of an existing custom view called ThrottledUserViewSet and renamed it to `FFAdminUserViewSet` to reflect our custom view and contain future customizations.